src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs

```diff
@@ -727,6 +727,9 @@ protected override void ProcessRecord()
                             Path = SessionState.Internal.GetSingleProvider(Commands.FileSystemProvider.ProviderName).Home;
                         }
 
+                        // Make sure we have a provider path and not PSPath.
+                        // PSPaths can be used, but break $pwd when used with SetLocation.
+                        Path = SessionState.Path.GetUnresolvedProviderPathFromPSPath(Path);
```

Comment:
Can you add tests for PSPath for the behavior you expect to work?
@anmenaga 	I think this is a breaking change.

Reply:
Ha! Yeah, when I fix it this'll be breaking. I'm focusing on one or two other PRs here but I'll get back to this one soon-ish.